Configurations.md

```diff
@@ -9,6 +9,12 @@ indent_style = "Block"
 reorder_imported_names = true
 ```
 
+Each configuration option is either stable or unstable.
+Stable options can be used directly, while unstable options are opt-in.
+
+To enable unstable options, set `unstable_features = true` in `rustfmt.toml` or pass `--unstable-options` to rustfmt,
+and ensure that the environment variable `CFG_RELEASE_CHANNEL` is set to `nightly`.
```

Comment:
This is not quite right - users need to be using a nightly toolchain, not setting that env var (which is a compile-time thing, iirc).

Reply:
I'm using nightly, but unless I set the env variable I still see these messages:

```
Warning: can't set `use_try_shorthand = true`, unstable features are only available in nightly channel.
Warning: can't set `error_on_line_overflow = false`, unstable features are only available in nightly channel.
Warning: can't set `error_on_line_overflow_comments = false`, unstable features are only available in nightly channel.
```

Reply:
I think I figured it out.

https://github.com/rust-lang-nursery/rustfmt/pull/2235 is a fix to inspect `CFG_RELEASE_CHANNEL` at rustfmt compile time, whereas in 0.2.17 which was published before that merge, it was checked at runtime. So the `rustfmt` distributed with the nightly toolchain will be built with support for unstable features, but the one distributed with the beta or stable toolchains won't.

What's the status of `CFG_RELEASE_CHANNEL` when building crates not as part of the toolchain? (e.g. `cargo +nightly install rustfmt-nightly`) The main advertised way of acquiring `cargo fmt` in the README is still via cargo, so we should make sure that it's possible to access the unstable configuration options from there.